### PR TITLE
IGAPP-1134: Add jsx-expressions/strict-logical-expressions linting rule

### PR DESCRIPTION
**Pull request by steffenkleinle**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Went for https://github.com/hpersson/eslint-plugin-jsx-expressions instead of https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-leaked-render.md because it supports ts types and therefore prevents false positives (e.g. for boolean props). Sadly, the rule only supports inline jsx, created an issue for that: https://github.com/hpersson/eslint-plugin-jsx-expressions/issues/11
Therefore still take care with jsx oustide of the return function.

### CODE REVIEW

**Review by LeandraH — APPROVED** [review]
Nice!

**Review by SteffiStuffel — DISMISSED** [review]
As asked in the lunes ticket:

Hmh, if the new plugin supports ts types someBoolean && <Text>... should be safe. As this is only not working, if the value is non-boolean or undefined, which is already prevented by ts-checks or am I missing something here?

File: web/src/components/PoiDetails.tsx [hunk 1]
```diff
@@ -202,21 +202,23 @@ const PoiDetails: React.FC<PoiDetailsProps> = ({
...
       </DetailSection>
-      {(website || phoneNumber || email) && (
+      {(!!website || !!phoneNumber || !!email) && (
         <>
           <Spacer borderColor={theme.colors.poiBorderColor} />
           <Collapsible title={t('contactInformation')} initialCollapsed direction={direction}>
             <>
-              {website && <ContactItem iconSrc={WebsiteIcon} iconAlt={t('website')} link={website} content={website} />}
-              {phoneNumber && (
+              {!!website && (
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply web/src/components/PoiDetails.tsx:210]: Just prettier because the line is now to long I guess.

**Review by f1sh1918 — APPROVED** [review]
looks fine to me

**Comment by steffenkleinle** [comment]
> As asked in the lunes ticket:
> 
> Hmh, if the new plugin supports ts types someBoolean && ... should be safe. As this is only not working, if the value is non-boolean or undefined, which is already prevented by ts-checks or am I missing something here?

See answer here: https://github.com/digitalfabrik/lunes-app/pull/384#issuecomment-1293265592

**Pull request merged**